server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -536,8 +570,21 @@ public ReplicationStats execute(Client client) throws Exception {
 
       // If we have some edits to send
       if (0 < edits.walEdits.getEditsSize()) {
+        // Check if we are interrupted before to writing the edits
+        if (Thread.interrupted()) {
+          log.debug("Replication work interrupted before writing edits, returning empty replication stats");
+          return new ReplicationStats(0L, 0L, 0L);
+        }
+
         log.debug("Sending {} edits", edits.walEdits.getEditsSize());
         long entriesReplicated = client.replicateLog(remoteTableId, edits.walEdits, tcreds);
```

Comment:
I don't see a difference either way.  What I am trying to address is a rare block in the replication pipeline, not necessarily that it takes long to do the work.  I'd just need the RPC call to timeout and bubble up so the tserver performing the replication work can release the lock so it is re-tried.

This does, however, complicate my testing since I am now dealing with Thrift timeouts.